airflow/operators/python.py

```diff
@@ -269,140 +281,132 @@ def __init__(  # pylint: disable=too-many-arguments
+            # ..but pendulum is installed so keep pendulum date objects
+            # Note: 'lazy_object_proxy' is needed to work.
+            return
+
+        # No pendulum is installed either. So remove pendulum specific context.
+        # otherwise "ModuleNotFoundError: No module named 'pendulum'" would be raised.
+        self.op_kwargs.pop('execution_date', None)
+        self.op_kwargs.pop('next_execution_date', None)
+        self.op_kwargs.pop('prev_execution_date', None)
+        self.op_kwargs.pop('prev_execution_date_success', None)
+        self.op_kwargs.pop('prev_start_date_success', None)
 
     def _write_string_args(self, filename):
-        # writes string_args to a file, which are read line by line
         with open(filename, 'w') as file:
             file.write('\n'.join(map(str, self.string_args)))
 
-    def _write_args(self, input_filename):
-        # serialize args to file
-        if self._pass_op_args():
-            with open(input_filename, 'wb') as file:
-                arg_dict = ({'args': self.op_args, 'kwargs': self.op_kwargs})
-                if self.use_dill:
-                    dill.dump(arg_dict, file)
-                else:
-                    pickle.dump(arg_dict, file)
-
-    def _read_result(self, output_filename):
-        if os.stat(output_filename).st_size == 0:
+    def _code__imports(self):
+        return f'import {self.pickling_library.__name__}\n' \
+               'import sys\n'
+
+    def _code__read_args(self, filename):
+        if self.op_args or self.op_kwargs:
+            return f'with open({filename}, "rb") as file:\n' \
+                   f'    arg_dict = {self.pickling_library.__name__}.load(file)\n'
+        return 'arg_dict = {"args": [], "kwargs": {}}\n'
+
+    def _code__read_string_args(self, filename):
+        return f'with open({filename}, "r") as file:\n' \
+               '    virtualenv_string_args = list(map(lambda x: x.strip(), list(file)))\n'
+
+    def _code__write_output(self, filename):
+        return f'with open({filename}, "wb") as file:\n' \
+               f'    if res: {self.pickling_library.__name__}.dump(res, file)\n'
+
+    def _code__call_script(self, arg_dict):
+        return f'{dedent(inspect.getsource(self.python_callable))}\n' \
+               f'res = {self.python_callable.__name__}(*{arg_dict}["args"], **{arg_dict}["kwargs"])\n'
+
+    def _write_script(self, filename):
+        with open(filename, 'w') as file:
+            python_code = f"{self._code__imports()}" \
+                          f"{self._code__read_args(filename='sys.argv[1]')}" \
+                          f"{self._code__read_string_args(filename='sys.argv[3]')}" \
+                          f"{self._code__call_script(arg_dict='arg_dict')}" \
+                          f"{self._code__write_output(filename='sys.argv[2]')}"
+            self.log.debug('Writing code to file\n %s', python_code)
+            file.write(python_code)
```

Comment:
@feluelle what would you say to use jinja template? I think it would be easier to understand how the `python_code` looks like. We used a similar approach in operator generator: 
https://github.com/PolideaInternal/airflow-munchkin/blob/master/airflow_munchkin/template/discovery/operator_class.tpl

Reply:
Looks good 👍 How do you use it then?

Reply:
We render the template and save it to a file.
https://github.com/PolideaInternal/airflow-munchkin/blob/2de8bff30b342f11e260dbfa8f7725c74731adda/airflow_munchkin/discovery_parser/renderers.py#L21-L25

https://github.com/PolideaInternal/airflow-munchkin/blob/2de8bff30b342f11e260dbfa8f7725c74731adda/airflow_munchkin/block_renderer/template_utils.py#L28-L32

I think being able to take a look at the whole structure of the file is quite helpful.

Reply:
Thanks - will take a look.

Reply:
Done @turbaszek 👍